### PR TITLE
Spanner

### PR DESCRIPTION
**Pull request by ccemeraldeyes**

### CODE REVIEW

File: brokerapi/brokers/brokers_test.go [hunk 1]
```diff
@@ -61,161 +62,12 @@ var _ = Describe("Brokers", func() {
...
-				"guid": "foo2",
-				"name": "bar2",
-				"description": "test-bigtable-plan",
-				"storage_type": "SSD",
-				"num_nodes": "3",
-				"display_name": "FOOBAR2",
-				"service": "b8e19880-ac58-42ef-b033-f7cd9c94d1fe"
-			}
-		}`)
+		os.Setenv("SERVICES", fakes.Services)
```
Comment (johnsonj) [inline brokerapi/brokers/brokers_test.go]: :+1:

File: brokerapi/brokers/gcp_service_broker.go [hunk 2]
```diff
@@ -589,6 +598,13 @@ func InitCatalogFromEnv() ([]models.Service, error) {
 	}
 	servicePlans[bigtableServiceId] = append(servicePlans[bigtableServiceId], bigtablePlans...)
 
+	// set up bigtable custom plans
```
Comment (johnsonj) [inline brokerapi/brokers/gcp_service_broker.go]: s/bigtable/spanner/

File: brokerapi/brokers/gcp_service_broker.go [hunk 3]
```diff
@@ -589,6 +598,13 @@ func InitCatalogFromEnv() ([]models.Service, error) {
...
+	spannerPlans, spannerServiceId, err := getDynamicPlans("SPANNER_CUSTOM_PLANS", spanner.MapPlan)
```
Comment (johnsonj) [inline brokerapi/brokers/gcp_service_broker.go]: did you consider extracting this custom plan save out to a helper method? It seems like it's general enough between the 3 but I could be wrong there. It's kinda the next step after your `getDynamicPlans` refactor.

I'd weigh if it was worth it here or not. Might be too dense to be worth it.

File: brokerapi/brokers/spanner/broker.go [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+	}
+
+	i := models.ServiceInstanceDetails{
+		Name:         params["name"],
+		Url:          "",
+		Location:     loc,
+		OtherDetails: string(otherDetails),
+	}
+
+	//op := client.InstanceOperation(params["name"])
```
Comment (johnsonj) [inline brokerapi/brokers/spanner/broker.go]: remove

File: brokerapi/brokers/spanner/broker.go [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+	ii := InstanceInformation{
+		InstanceId: params["name"],
+	}
+
+	otherDetails, err := json.Marshal(ii)
+	if err != nil {
+		return models.ServiceInstanceDetails{}, fmt.Errorf("Error marshalling other details: %s", err)
+	}
+
+	i := models.ServiceInstanceDetails{
```
Comment (johnsonj) [inline brokerapi/brokers/spanner/broker.go]: Do you think it'd be useful to move the update of `ServiceInstanceDetail` into `BrokerBase`? All of the brokers I am familiar with do this serialize otherDetails then persist this model.

File: brokerapi/brokers/spanner/broker.go [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+	return false, fmt.Errorf("unknown error")
+}
+
+func createCloudOperation(op *googlespanner.CreateInstanceOperation, instanceId string, serviceId string) error {
+	var err error
+
+	_, err = op.Poll(context.Background())
+
+	errorStr := ""
+	if err != nil {
```
Comment (johnsonj) [inline brokerapi/brokers/spanner/broker.go]: another way to do this, since you don't care about the first return of `op.Poll`:

```go
errorStr := ""
if _. err := op.Poll(context.background(); err != nil {
   errorStr := err.Error()
}
```

Then no var error and it's a bit more compact.

File: examples/go/sb-test/api.go [hunk 7]
```diff
@@ -302,15 +326,20 @@ func pullFromPubSub(w http.ResponseWriter, req *http.Request) {
 	}
 	pubsubService, _ := pubsub.NewClient(context.Background(), projectId, option.WithHTTPClient(conf.Client(context.Background())))
 
-	it, err := pubsubService.Subscription("projects/" + projectId + "/subscriptions/" + subscriptionName).Pull(context.Background())
+	cctx, cancel := context.WithCancel(context.Background())
+	//   err := sub.Receive(cctx, callback)
```
Comment (johnsonj) [inline examples/go/sb-test/api.go]: remove

File: brokerapi/brokers/gcp_service_broker.go (hunk 2 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/gcp_service_broker.go]: check.

File: brokerapi/brokers/gcp_service_broker.go (hunk 3 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/gcp_service_broker.go]: absolutely agree that this bit should be refactored, but I want to do it as its own PR.

File: brokerapi/brokers/spanner/broker.go (hunk 4 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/spanner/broker.go]: check.

File: brokerapi/brokers/spanner/broker.go (hunk 6 above)
Comment (ccemeraldeyes) [inline brokerapi/brokers/spanner/broker.go]: check.

File: examples/go/sb-test/api.go (hunk 7 above)
Comment (ccemeraldeyes) [inline examples/go/sb-test/api.go]: check
